### PR TITLE
fix: Update launch_args env.

### PR DESCRIPTION
**Pull request by prmukherj**
import ansys.fluent.core as pyfluent
s = pyfluent.launch_fluent(ui_mode="gui", custom_fluent_path=r"...\vNNN\fluent\ntbin\win64\fluent.exe")

### CODE REVIEW

**Review by hpohekar — APPROVED** [review]

**Review by tejalprabhu01 — APPROVED** [review]
Tested the changes! works as expected!

**Comment by hpohekar** [comment]
@prmukherj @seanpearsonuk @mkundu1 

As per [the previous discussion](https://github.com/ansys/pyfluent/issues/2791) `PYFLUENT_FLUENT_ROOT` is for local development. It is not documented for users. User should specify `product_version` to launch specific Fluent version not the `PYFLUENT_FLUENT_ROOT`.

**Comment by tejalprabhu01** [comment]
> @prmukherj @seanpearsonuk @mkundu1
> 
> As per [the previous discussion](https://github.com/ansys/pyfluent/issues/2791) `PYFLUENT_FLUENT_ROOT` is for local development. It is not documented for users. User should specify `product_version` to launch specific Fluent version not the `PYFLUENT_FLUENT_ROOT`.

@hpohekar Arent users allowed to provide any custom paths to Fluent installation (other than AWP_ROOT)? I am asking because in optiSLang we provide users setting where they can provide any path to fluent installation. If this should not be the case; then we might have to also do the changes on optiSLang side.

**Comment by prmukherj** [comment]
> > @prmukherj @seanpearsonuk @mkundu1
> > As per [the previous discussion](https://github.com/ansys/pyfluent/issues/2791) `PYFLUENT_FLUENT_ROOT` is for local development. It is not documented for users. User should specify `product_version` to launch specific Fluent version not the `PYFLUENT_FLUENT_ROOT`.
> 
> @hpohekar Arent users allowed to provide any custom paths to Fluent installation (other than AWP_ROOT)? I am asking because in optiSLang we provide users setting where they can provide any path to fluent installation. If this should not be the case; then we might have to also do the changes on optiSLang side. Based on the discussion link you provided, it still isnt clear to me.

Ideally providing a custom path should be allowed. We'll discuss this within the team and get back.
Thanks

**Comment by hpohekar** [comment]
@tejalprabhu01 

Aren't users allowed to provide any custom paths to Fluent installation (other than AWP_ROOT)? - No.

User don't need to specify any custom paths on Windows as `AWP_ROOT<version>` is already set for each Fluent release. So user just need to specify product version.

On Linux, user need to specify custom Fluent version path using AWP_ROOT<version> only.

Please have a look at the following specification.

image

https://fluent.docs.pyansys.com/version/stable/getting_started/installation.html

**Comment by tejalprabhu01** [comment]
@hpohekar @seanpearsonuk @mkundu1 
There will be no AWP_ROOT when using Fluent installation from shared location and also we have previously encountered that users sometimes do not have administrative rights and hence cannot set the environment variables like AWP_ROOT. I understand having to set only the version makes things easier for the pyfluent consumers. But it is also important to have a bit of flexibility when using pyfluent from other tools like optiSLang.

**Comment by hpohekar** [comment]
> @hpohekar @seanpearsonuk @mkundu1 There will be no AWP_ROOT when using Fluent installation from shared location and also we have previously encountered that users sometimes do not have administrative rights and hence cannot set the environment variables like AWP_ROOT. I understand having to set only the version makes things easier for the pyfluent consumers. But it is also important to have a bit of flexibility when using pyfluent from other tools like optiSLang.

@tejalprabhu01 @seanpearsonuk @mkundu1 

Ok. I see. We should document `PYFLUENT_FLUENT_ROOT` usage considering cases like use of Fluent installation from shared location and lack of administrative rights to set environment variables. 

It is worth of considering the point of lack of administrative rights to set environment variable, specifically on Linux system because current PyFluent document suggests to set AWP_ROOT compulsory.

Thanks for the clarification.

**Comment by mkundu1** [comment]
Note that, the `env` argument in `launch_fluent` is for setting environment variables in the Fluent process. It is better not to mix PyFluent process's env vars within it. For PyFluent, `os.environ` or a new `launch_fluent` argument for setting custom Fluent path is a better solution.
@prmukherj @hpohekar

**Comment by prmukherj** [comment]
> Note that, the `env` argument in `launch_fluent` is for setting environment variables in the Fluent process. It is better not to mix PyFluent process's env vars within it. For PyFluent, `os.environ` or a new `launch_fluent` argument for setting custom Fluent path is a better solution. @prmukherj @hpohekar

I agree with this point.
Thank you Mainak. Will just confirm from @seanpearsonuk and @hpohekar about adding this new argument -> 'custom_fluent_path'?

Thank you.

**Comment by hpohekar** [comment]
> Note that, the `env` argument in `launch_fluent` is for setting environment variables in the Fluent process. It is better not to mix PyFluent process's env vars within it. For PyFluent, `os.environ` or a new `launch_fluent` argument for setting custom Fluent path is a better solution. @prmukherj @hpohekar

@mkundu1 Yes, I agree. Thank you.

File: src/ansys/fluent/core/launcher/slurm_launcher.py [hunk 1]
```diff
@@ -281,6 +281,7 @@ def __init__(
         py: Optional[bool] = None,
         gpu: Optional[bool] = None,
         cwd: Optional[str] = None,
+        custom_fluent_path: Optional[str] = None,
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/launcher/slurm_launcher.py:284]: @prmukherj Is this argument appropriate here?

Comment (prmukherj) [reply src/ansys/fluent/core/launcher/slurm_launcher.py:284]: @mkundu1, could you please confirm if this is required for Slurm launcher?

Comment (seanpearsonuk) [reply src/ansys/fluent/core/launcher/slurm_launcher.py:284]: @prmukherj we could go ahead and merge meanwhile.

**Review by hpohekar — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Review by tejalprabhu01 — APPROVED** [review]
Thanks for the implementation.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -452,7 +452,7 @@ jobs:
     name: Unit Testing
     if: ${{ always() }}
     needs: build
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, pyfluent]
```
Comment (prmukherj) [inline .github/workflows/ci.yml:455]:
```suggestion
    runs-on: [public-ubuntu-latest-8-cores]
```

File: src/ansys/fluent/core/launcher/launcher.py [hunk 3]
```diff
@@ -117,6 +117,7 @@ def launch_fluent(
     py: Optional[bool] = None,
     gpu: Union[bool, list[int], None] = None,
     cwd: Optional[str] = None,
+    custom_fluent_path: Optional[str] = None,
```
Comment (mkundu1) [inline src/ansys/fluent/core/launcher/launcher.py:120]: Can be replaced with a shorter name `fluent_path`

Comment (prmukherj) [inline src/ansys/fluent/core/launcher/launcher.py:120]:
```suggestion
    fluent_path: Optional[str] = None,
```

File: src/ansys/fluent/core/launcher/launcher.py [hunk 4]
```diff
@@ -217,6 +218,8 @@ def launch_fluent(
         information like how to determine the GPU IDs.
     cwd : str, Optional
         Working directory for the Fluent client.
+    custom_fluent_path: str, Optional
```
Comment (prmukherj) [inline src/ansys/fluent/core/launcher/launcher.py:221]:
```suggestion
    fluent_path: str, Optional
```

File: src/ansys/fluent/core/launcher/process_launch_string.py [hunk 5]
```diff
@@ -150,5 +150,10 @@ def get_exe_path(fluent_root: Path) -> Path:
     if fluent_root:
         return get_exe_path(Path(fluent_root))
 
-    # 2. the latest ANSYS version from AWP_ROOT environment variables
+    # 2. Custom Path provided by the user in launch_fluent
+    fluent_path = launch_argvals.get("custom_fluent_path")
```
Comment (prmukherj) [inline src/ansys/fluent/core/launcher/process_launch_string.py:154]:
```suggestion
    fluent_path = launch_argvals.get("fluent_path")
```

File: src/ansys/fluent/core/launcher/slurm_launcher.py (hunk 1 above)
Comment (prmukherj) [inline src/ansys/fluent/core/launcher/slurm_launcher.py:284]:
```suggestion
        fluent_path: Optional[str] = None,
```

File: src/ansys/fluent/core/launcher/slurm_launcher.py [hunk 6]
```diff
@@ -353,6 +354,8 @@ def __init__(
             If True, Fluent will start with GPU Solver.
         cwd : str, Optional
             Working directory for the Fluent client.
+        custom_fluent_path: str, Optional
```
Comment (prmukherj) [inline src/ansys/fluent/core/launcher/slurm_launcher.py:357]:
```suggestion
        fluent_path: str, Optional
```

File: src/ansys/fluent/core/launcher/standalone_launcher.py [hunk 7]
```diff
@@ -78,6 +78,7 @@ def __init__(
         py: Optional[bool] = None,
         gpu: Optional[bool] = None,
         cwd: Optional[str] = None,
+        custom_fluent_path: Optional[str] = None,
```
Comment (prmukherj) [inline src/ansys/fluent/core/launcher/standalone_launcher.py:81]:
```suggestion
        fluent_path: Optional[str] = None,
```

File: src/ansys/fluent/core/launcher/standalone_launcher.py [hunk 8]
```diff
@@ -149,6 +150,8 @@ def __init__(
             If True, Fluent will start with GPU Solver.
         cwd : str, Optional
             Working directory for the Fluent client.
+        custom_fluent_path: str, Optional
```
Comment (prmukherj) [inline src/ansys/fluent/core/launcher/standalone_launcher.py:153]:
```suggestion
        fluent_path: str, Optional
```

**Pull request merged**